core/src/main/java/org/elasticsearch/index/shard/ShardStateMetaData.java

```diff
@@ -41,15 +41,19 @@
     private static final String INDEX_UUID_KEY = "index_uuid";
     private static final String ALLOCATION_ID_KEY = "allocation_id";
 
-    public final long version;
+    public final long legacyVersion; // for pre-3.0 shards that have not yet been active
     public final String indexUUID;
     public final boolean primary;
     @Nullable
     public final AllocationId allocationId; // can be null if we read from legacy format (see fromXContent and MultiDataPathUpgrader)
 
-    public ShardStateMetaData(long version, boolean primary, String indexUUID, AllocationId allocationId) {
+    public ShardStateMetaData(boolean primary, String indexUUID, AllocationId allocationId) {
+        this(-1L, primary, indexUUID, allocationId);
```

Comment:
can we make that -1 a constant and use it in all the relevant places? it would be easier to remove it once we go to 4.0